src/components/enterprise/dashboard/sidebar/offers/Offer.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+
+import './styles/Offer.scss';
+
+class Offer extends React.Component {
+  renderTitle(usageType, benefitValue) {
+    if (usageType === 'Percentage') {
+      if (benefitValue === 100) {
+        return 'Enroll for free';
+      }
+      return `Get ${benefitValue}% off`;
+    }
+    return `Get $${benefitValue} off`;
+  }
+
+  renderFinePrint(redemptionsRemaining, couponEndDate) {
+    let message = `Expires ${moment(couponEndDate).format('MMMM D, YYYY')}.`;
+    if (redemptionsRemaining > 1) {
+      message = `You can use this ${redemptionsRemaining} more times. ${message}`;
+    }
+    return message;
+  }
+
+  render() {
+    const {
+      usageType,
+      benefitValue,
+      redemptionsRemaining,
+      code,
+      couponEndDate,
+    } = this.props;
+    const offerUrl = `${process.env.ECOMMERCE_BASE_URL}/coupons/offer/?code=${code}`;
+
+    return (
+      <a href={offerUrl} className="offer card mb-3 d-block">
+        <div className="card-body">
+          <h5 className="card-title font-weight-bold h6">
+            {this.renderTitle(usageType, benefitValue)}
+          </h5>
+          <p className="card-text">
+            {this.renderFinePrint(redemptionsRemaining, couponEndDate)}
+          </p>
+        </div>
+      </a>
+    );
+  }
+}
+
+Offer.defaultProps = {
+  usageType: null,
+  benefitValue: null,
+  redemptionsRemaining: null,
+  code: null,
+  couponEndDate: null,
+};
+
+Offer.propTypes = {
+  usageType: PropTypes.string,
```

Comment:
Should we make all of these required props, and leave of the `null` default props? If the component always expects those props to be there for rendering, probably makes sense to have them required.